pyomo/contrib/mindtpy/initialization.py

```diff
@@ -15,13 +15,18 @@
 from pyomo.contrib.mindtpy.nlp_solve import (solve_NLP_subproblem,
                                              handle_NLP_subproblem_optimal, handle_NLP_subproblem_infeasible,
                                              handle_NLP_subproblem_other_termination)
+from pyomo.contrib.mindtpy.single_tree import var_bound_add
```

Comment:
This unconditionally imports `single_tree`, which in turn unconditionally imports cplex.  This leads to failures on platforms where the cplex python bindings are not present.

Reply:
@bernalde w.r.t. your question about how to handle this. You want to make sure that your symbols are properly scoped. Otherwise you're going to run yourself into an import loop someday. I would argue that `var_bound_add` should be moved to a more general utility module.